packages/dds/tree/src/test/time/tableTree.spec.ts

```diff
@@ -0,0 +1,646 @@
+	title,
+	tableSize,
+	cellValue,
+	setupOperation,
+	stackOperation,
+	minBatchDurationSeconds = 0,
+	maxBenchmarkDurationSeconds,
+}: {
+	title: string;
+	tableSize: number;
+	cellValue: string;
+	/**
+	 * A function that sets up the operation to be performed on the tree.
+	 */
+	setupOperation: (tree: TableTreeDefinition) => void;
+	/**
+	 * The operation to perform on the stack. This should be a function that takes an UndoRedoStackManager
+	 * and performs the desired operation.
+	 */
+	stackOperation: (tree: TableTreeDefinition) => void;
+	minBatchDurationSeconds?: number;
+	maxBenchmarkDurationSeconds: number;
+}) {
+	benchmark({
+		type: BenchmarkType.Measurement,
+		title,
+		benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
+			let duration: number;
+			do {
+				assert.equal(state.iterationsPerBatch, 1, "Expected exactly one iteration per batch");
+
+				// Setup
+				const localTree: TableTreeDefinition = createTableTree(tableSize, cellValue);
+				setupOperation(localTree);
+
+				// Operation
+				const before = state.timer.now();
+				stackOperation(localTree);
+				const after = state.timer.now();
+
+				// Measure
+				duration = state.timer.toSeconds(before, after);
+
+				// Cleanup
+				localTree.unsubscribe();
+			} while (state.recordBatch(duration));
+		},
+		minBatchDurationSeconds,
+		maxBenchmarkDurationSeconds,
+	});
+}
+
+describe("SharedTree execution time", () => {
+	// The value to be set in the cells of the tree.
+	const cellValue = "cellValue";
+	// The test tree's size will be 10*10, 100*100.
+	// Tree size 1000 benchmarks removed due to high overhead and unreliable results.
+	const treeSizes = isInPerformanceTestingMode
+		? [10, 100]
+		: // When not measuring perf, use a single smaller data size so the tests run faster.
+			[10];
+
+	// The number of operations to perform on the tree.
+	// Operation counts 1000 removed due to high overhead and unreliable results.
+	const operationCounts = isInPerformanceTestingMode
+		? [10, 100]
+		: // When not measuring perf, use a single smaller data size so the tests run faster.
+			[5];
+	let maxBenchmarkDurationSeconds: number;
+
+	for (const treeSize of treeSizes) {
+		maxBenchmarkDurationSeconds = treeSize === 100 ? 10 : 5;
+
+		// Filter counts to ensure remove operation do not exceed treeSize
+		const validRemoveCounts = operationCounts.filter((count) => count <= treeSize);
+
+		// Insert-related tests that are not limited by treeSize
+		for (const count of operationCounts) {
+			describe(`Column Insertion`, () => {
+				// Test the execute time of the SharedTree for inserting a column in the middle for a given number of times.
+				runBenchmark({
+					title: `Insert a column in the middle ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					operation: (tree: TableTreeDefinition) => {
+						const { Column, table } = tree;
+						for (let i = 0; i < count; i++) {
+							const column = new Column({ id: `column-${uuid()}` });
```

Comment:
Nit: I would recommend not specifying the `id` property for columns and rows created in these tests. If you omit them, SharedTree will generate them automatically, which we expect to be the more common usage pattern.

Reply:
Remove `id`